sematic/api/endpoints/resolutions.py

```diff
@@ -261,3 +263,21 @@ def cancel_resolution_endpoint(
     )
 
     return flask.jsonify(dict(content=resolution.to_json_encodable()))
+
+
+@sematic_api.route(
+    "/api/v1/resolutions/<resolution_id>/external_resources", methods=["GET"]
+)
+@authenticate
+def get_resources_endpoint(user: Optional[User], resolution_id: str) -> flask.Response:
+    resource_ids = get_resource_ids_by_root_id(resolution_id)
+    resources = []
+    for resource_id in resource_ids:
+        resource = get_external_resource_record(resource_id)
```

Comment:
Are these N+1 requests still necessary? Can we just do `get_external_resources_by_root_id` and use a JOIN?

Reply:
Doing it in the same query as the JOIN would result in duplicate `ExternalResourceRecord` objects being returned from the DB in the case of multiple runs using a resource (the JOIN would have one row for each run/resource pair). But instead of N+1 I can make it 2: 1 to get the ids (which we then deduplicate), and 1 to get the objects.

Reply:
I believe this might be solvable using `distinct`. Added a suggestion.